adr/keycloak-adr.md

```diff
@@ -0,0 +1,236 @@
+#### Architectural Decision Record (ADR): Integration of Keycloak with GitLab for Authentication
+
+#### Status
+Proposed
+
+#### Date
+2024-03-05
+
+
+#### Introduction
+This ADR supports the further integration of Keycloak with GitLab to streamline authentication mechanisms, with the goal of enhancing security and user experience. The drive behind this initiative is to refine login processes, bolster security protocols, and ensure compliance with Information Assurance (IA) standards.
+
+**Objective**: Utilize Keycloak as the predominant Identity and Access Management (IAM) solution alongside GitLab, applying OAuth2 for authentication. This plan aims to consolidate user management and authentication, making credential management more straightforward and improving our security landscape.
```

Comment:
```suggestion
**Objective**: Utilize Keycloak as the predominant Identity and Access Management (IAM) solution alongside GitLab, applying OAuth2 for authentication. This plan aims to consolidate user management and authentication, making credential management more straightforward and improving our security landscape.
```
Would like to see the pros/cons and tradeoffs behind the decision to go with OAuth2 vs SAML initially (given SAML will give us more security features/support within the environment)